Store sequence metadata locally

Vectornaut opened the pull request:
### Summary

The branch to be merged uses local sequence file reads as drop-in replacements for OEIS searches.

### Motivation

This change paves the way for code simplifications and reliability improvements. For example, it makes sequences' names and raw references consistently fast to fetch, putting us in a position to fetch them synchronously when a sequence is requested.

This change also makes issue #161 mostly moot. It does this by eliminating the need for OEIS searches from every endpoint except `search_oeis`. The `search_oeis` endpoint is still broken on the branch to be merged, since we don't actually address the search format change.

### Documentation and manual testing

Before you can test the new code, you'll need to download the OEIS metadata into the right place. You can find instructions for that in the [quick-start guide](../../../Vectornaut/backscope/blob/local-metadata/README.md#quick-start) and the [Ubuntu installation instructions](../../../Vectornaut/backscope/blob/local-metadata/doc/install-ubuntu.md#clone-backscope) on the branch to be merged.

The database schema hasn't changed, so you can run the new code without wiping out the database structure. In fact, you don't even need to clear the stored data, although doing that might make it easier to reason about what's going on during manual testing.

### Automated testing

The tests on the branch to be merged use sequences' real names instead of placeholders, because names and raw references are now written to the database almost immediately after `fetch_metadata` is called. Since we're still fetching names and raw references asynchronously, this could make the tests flaky. Once we switch to fetching names and raw references synchronously, the tests should be come reliable again.

katestange commented:
I cleared my database and I tried using it via my frontscope and I was able to call up various random sequences.   Including A001122 which I think was one of the previously broken sequences on the server.  So that's great!  But the metadata the frontscope is showing for that one is strange.  Whether it pulls up the correct description seems 50-50.  Here are some failed examples:
![Screenshot from 2025-02-06 09-48-02](https://github.com/user-attachments/assets/f9f59867-0e2c-4f67-8e75-0fbb113ebdf1)
![Screenshot from 2025-02-06 09-47-19](https://github.com/user-attachments/assets/75a1339d-9f36-4a05-921b-07a46bae10b6)
![Screenshot from 2025-02-06 09-47-06](https://github.com/user-attachments/assets/ee920bfd-fe1e-4d21-a191-2f0b5ca4c34c)

Vectornaut commented:
The `get_oeis_metadata` endpoint should work, so this is a good catch!

Vectornaut commented:
> I cleared my database and I tried using it via my frontscope and I was able to call up various random sequences. Including A001122 which I think was one of the previously broken sequences on the server. So that's great! But the metadata the frontscope is showing for that one is strange.

I fixed this in commit cf0ec7e by removing the stopgap name, like Glen suggested. It seems like this problem should've existed before we started developing the local metadata feature, so it would be useful to know why we didn't notice it before. This is hard to investigate, though, because the OEIS search format change prevents the old code from working as-is.

Vectornaut commented:
I've added `ripgrep` as a dependency (in commit 2fa6d6c), so @katestange should be able to call `get_oeis_metadata` successfully after running `install-requirements.sh`.

Vectornaut commented:
I noticed by chance that when I rewrote `get_oeis_name_and_values`, I introduced a bug by forgetting to strip the sequence name. Commit 2c59e77 fixes the bug and adds a test that would've caught it.

katestange reviewed (approved):
So I've been hitting this pretty hard with frontscope in the process of testing the color PR.  All the endpoints seem to work now (except search as expected).  I have no further suggestions!  It would be great to put this on the server!

katestange commented:
@gwhitney @Vectornaut should I squash and merge?

gwhitney commented:
I was going to squash and merge for you, but then I noticed that it should be pretty easy to restore at least some functionality for the search_oeis endpoint. So I made that change, so Kate, I think it would be great if you could try that and if it at least does something for you, go ahead and squash and merge. Let me know when you've done that and I will install on the server (possibly updating server install instructions as I do).

katestange commented:
Works!  Cool!  Merging.
